src/modules/variable/set.rs

```diff
@@ -65,23 +75,19 @@ impl TranslateModule for VariableSet {
     fn translate(&self, meta: &mut TranslateMetadata) -> String {
         let name = self.name.clone();
         let index = self.index.as_ref()
-            .map(|index| format!("[{}]", self.is_ref
-                .then(|| index.translate_eval(meta, true))
-                .unwrap_or_else(|| index.translate(meta))))
+            .map(|index| self.translate_eval_if_ref(index, meta))
```

Comment:
Add helper function `translate_eval_if_ref()` to remove duplicated code here. I would have moved this new function to the definition of `TranslateModule`, but it only seems to be required here.